Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -553,6 +553,10 @@ public void Inline(Program program)
 
       var processedProgram = PreProcessProgramVerification(program);
 
+      if (Options.UseResolvedProgram.Any()) {
+        Options.UseResolvedProgram.ForEach(action => action(Options, processedProgram));
```

Comment:
I would use a C# foreach construct here, but it's a bit of style thing. One thing is that it's nicer to step through the iterations with a debugger. Another thing is that IDEs will be able to analyze it better.